Fix OSX builds

Tobias-Fischer opened the pull request:
See https://github.com/ros/geometry2/pull/502 for reasoning. This was confirmed to work in the RoboStack project: https://github.com/RoboStack/ros-noetic/issues/58

Tobias-Fischer committed:
Fix OSX builds

jacobperron commented on CMakeLists.txt:
```diff
@@ -7,11 +7,19 @@ set(Python_ADDITIONAL_VERSIONS "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}"
-  LIBRARIES rospack ${PYTHON_LIBRARIES}
-  DEPENDS Boost TinyXML2
-)
+if(NOT APPLE)
+  catkin_package(
+    INCLUDE_DIRS include
+    LIBRARIES rospack ${PYTHON_LIBRARIES}
+    DEPENDS Boost TinyXML2
+  )
+else()
+  catkin_package(
+    INCLUDE_DIRS include
+    LIBRARIES rospack
+    DEPENDS Boost TinyXML2
+ )
+endif()
```
To help reduce duplication, I think it would be nicer to do something like this:

```cmake
set(package_libraries rospack)
if(NOT APPLE)
  list(APPEND package_libraries ${PYTHON_LIBRARIES}
endif()

catkin_package(
  INCLUDE_DIRS include
  LIBRARIES ${package_libraries}
  DEPENDS Boost TinyXML2
)
```

jacobperron commented on CMakeLists.txt:
```diff
@@ -29,7 +37,17 @@ add_library(rospack
 )
-target_link_libraries(rospack ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
+
+if(NOT APPLE)
+  target_link_libraries(rospack ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
+else()
+  target_link_libraries(rospack ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES})
+endif()
```
Same here, something like

```cmake
set(link_libraries ${TinyXML2_LIBRARIES} ${Boost_LIBRARIES})
if(NOT APPLE)
  list(APPEND link_libraries ${PYTHON_LIBRARIES})
endif()
target_link_libraries(rospack ${link_libraries})
```

jacobperron reviewed (commented):
Some minor suggestions, otherwise LGTM!

Tobias-Fischer committed:
Incorporate suggestions by @jacobperron

Tobias-Fischer committed:
Update CMakeLists.txt

Tobias-Fischer commented:
Thanks for looking at that @jacobperron. I incorporated your suggestion. The link to PYTHON_LIBRARIES was actually needed, so I removed this part from the PR. We confirmed that rospack now works as intended in the RoboStack project: https://github.com/RoboStack/ros-noetic

jacobperron commented on CMakeLists.txt:
```diff
@@ -29,6 +34,7 @@ add_library(rospack
   src/rospack_cmdline.cpp
   src/utils.cpp
 )
+
```
nit: remove unrelated whitespace change to keep diff minimal

jacobperron commented on CMakeLists.txt:
```diff
@@ -7,9 +7,14 @@ set(Python_ADDITIONAL_VERSIONS "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}"
 
+set(package_libraries rospack)
+if(NOT APPLE)
+  list(APPEND package_libraries ${PYTHON_LIBRARIES}
```
```suggestion
  list(APPEND package_libraries ${PYTHON_LIBRARIES})
```

Tobias-Fischer committed:
Update CMakeLists.txt

Co-authored-by: Jacob Perron <jacob@openrobotics.org>

Tobias-Fischer committed:
Update CMakeLists.txt

Tobias-Fischer commented:
Done.

Tobias-Fischer commented:
Thanks for merging @jacobperron!
